### ISSUE TITLE
Triggering a cancellation token when calling ShowMessageDialogAsync doesn't close the popup on Skia

### ISSUE THREAD

**Issue by HakanL** [unoplatform/uno.extensions#1539]
## Current behavior

When displaying a modal dialog box and triggering the cancel even the dialog doesn't close on Skia. It works in WinUI. The method is returning, but the dialog is still displaying. It closes if I click the button (mouse/touch), but I need to close the dialog using the cancel event (timer or external hardware that triggers the cancel event).

## Expected behavior

Dialog should return (which it does), **and close** the dialog UI.

## How to reproduce it (as minimally and precisely as possible)

```
var cts = new CancellationTokenSource(TimeSpan.FromSeconds(5));

await this.navigator.ShowMessageDialogAsync(this, title: "Dialog", content: "I should disappear after 5 seconds", cancellation: cts.Token);
```

## Environment

Nuget Package: Uno.Extensions.Navigation.WinUI (I assume)

Package Version(s): 2.4.1

Affected platform(s):

- [X] Skia GTK Windows
- [X] Skia Linux FB
- [X] Windows

Visual Studio:

- [X] 2022

## Anything else we need to know?

Possibly related to https://github.com/unoplatform/uno.extensions/issues/1480, however in this scenario I'm using the **built-in dialog** so I don't have control over using ContentDialog vs Page.

**Comment by nickrandolph** [comment]
Thanks for reporting - I'll review both this and #1480 again today and try to get a fix for you

**Comment by nickrandolph** [comment]
Hey @HakanL can you confirm which target platform you're seeing this on - if on GTK are you running on Windows or linux via WSL?
Asking because I just ran the Skia-GTK head of the test harness and ran this test (https://github.com/unoplatform/uno.extensions/blob/180ecea87b49e3bcd3225b06b85c4ab64be07b63/testing/TestHarness/TestHarness.Shared/Ext/Navigation/Dialogs/MessageDialogsPage.xaml.cs#L26-L30)
and the message dialog did close as expected after the 2 seconds when the token is cancelled

**Comment by HakanL** [comment]
GTK Skia on Windows, and Skia/LinuxFB on RPi is where I'm running this, and I see the issue in both places. I haven't tested the test harness, but the call to ShowMessageDialog exits as expected, but the dialog is still visible on the screen, does the harness test for that?

**Comment by nickrandolph** [comment]
I can't recall if the actual UI tests validate this but I'm running the test harness solution and triggering it manually. After 2 seconds the dialog disappears. Is your application now targeting latest uno stable version? If so, I'll try updating the harness, as I think it's referencing an older version, so perhaps there's a regression in uno.

**Comment by HakanL** [comment]
@nickrandolph Oh, odd, if I set the cancel as a timer, then it disappears, but if I trigger it externally then it doesn't work. Note that my RPi hardware has a push button that will work like a select button, so it should close the dialog box.

**Comment by HakanL** [comment]
I'm on 2.4.2, latest stable as of a few minutes ago.

**Comment by HakanL** [comment]
I'm wondering if it could be related to threading somehow (synchronization context or something).

**Comment by nickrandolph** [comment]
ok, let me try triggering the cancellation via a different mechanism (eg background thread)

**Comment by HakanL** [comment]
(in my example in the bug report I used a timer, because I assume that would be identical to external trigger), my bad.

**Comment by nickrandolph** [comment]
even if I create a background task and cancel from that it still closes the dialog. Let me switch to latest uno stable and confirm no regression

**Comment by HakanL** [comment]
Ok, so now when I'm on 2.4.2 it actually works with my external trigger as well. Could this have been fixed? I may have been on 2.5-dev66 earlier, I can't remember now (too many things going on)

**Comment by nickrandolph** [comment]
I think it's more likely that a change in Uno would have fixed this - I know there's on-going investment in the skia targets, so it wouldn't surprise me that it's fixed there. 
If you can double check this is working and then close the issue if it's all good now

**Comment by HakanL** [comment]
Ok, so this was stupid, long story short, I'm using a rotary encoder with push button to navigate my UI on a RPi (so I move the keyboard focus fwd/bwd using the rotary knob, then push to select). I also use the push button to close the dialog box. Well, it turns out I tested the dialog box by selecting an item and then pushing the push button to trigger the dialog. Then a push again would close the dialog. What happened was that the second push (to close the dialog) actually then immediately triggered the dialog again (since focus was still on the menu item that triggers the test dialog).... So, my bad, sorry to waste your time, I should've added a log entry :)
I'm still not sure about #1480, it wasn't the same scenario (I'm using that dialog/page to prevent touches since I turn off the backlight, I think I discussed it here but I can't find the reference, basically I tried to hook into the PointerPressed and setting e.Handled = true so it wouldn't bubble up, but it didn't work so instead it was suggested that I display a fullscreen dialog that captures all touches until the backlight is back on).
